packages/manager/src/features/Betas/BetaDetails.tsx

```diff
@@ -63,7 +63,7 @@ const BetaDetails = (props: Props) => {
         {enrolled ? null : (
           <Button
             onClick={() => {
-              history.push('/betas/signup', { betaId: id });
+              navigate({ to: `/betas/signup/${id}` });
```

Comment:
This might be more a preferable way to write this, but I'm not 100% sure what difference it makes. They might both be okay and equally as typesafe

```suggestion
              navigate({ params: { betaId: id }, to: '/betas/signup/$betaId' });
```

Reply:
i think you are right, from looking at this more., looks like the approach with params uses a parameterized route path ('/betas/signup/$betaId') and provides the parameter separately in the params object. This allows TanStack Router to perform type checking on the parameters, ensuring that you're providing the correct parameters for the route. It also avoids string Interpolation which can be error-prone and doesn't provide type checking for the interpolated values.